File: 3379-transformed-array/3379-transformed-array.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        size = len(nums)
        nums = nums[::-1] + nums + nums
        res = []
        for i in range(size, 2 * size):
            res.append(nums[i + (nums[i] % size)])
        
        return res

File: 3379-transformed-array/3379-transformed-array.go
func constructTransformedArray(nums []int) []int {
    n := len(nums)
    res := make([]int, n)
    for i, val := range nums {
        index := (((i + val) % n) + n) % n
        res[i] = nums[index]
    }
    return res
}